### ISSUE TITLE
Bug in @solana/errors

### ISSUE THREAD

**Issue by lukecaan** [solana-labs/solana-web3.js#2903]
Tldr; a bad regex string in @solana/errors is makes old (<16.4) iOS browsers crash

..

Hi, the same issue which was reported here https://github.com/solana-labs/solana-web3.js/pull/2745, and ended up being fixed here https://github.com/solana-labs/solana-web3.js/pull/2785/files is rearing its head again.

The issue this time appears to be `@solana/errors`, which is a nested dependency of `@solana/spl-token` > `@solana/spl-token-metadata` > `@solana/solana-codecs`  > `@solana/solana/errors`

It looks like the latest version of the code in `packages/errors` should actually be fine - but if you pull the latest bundled code from npm it is still using the broken version of `getHumanReadableErrorMessage`

<img width="761" alt="image" src="https://github.com/solana-labs/solana-web3.js/assets/14064888/dec3e9a0-7fad-4b29-bdf7-b8ad11ec41c9">

^ The negative lookbehind regex is the problem. You can see the broken version of the method in the npm bundled code if you go here https://www.npmjs.com/package/@solana/errors/v/2.0.0-preview.3?activeTab=code and look inside`dist/cli.js`

I think doing a few fresh releases making things look at newer package versions will fix it

cc @steveluscher

**Comment by steveluscher** [comment]
Super coincidence, but we're doing a release of TP4 today!

**Comment by steveluscher** [comment]
https://github.com/solana-labs/solana-web3.js/releases/tag/tp4

**Comment by lukecaan** [comment]
@steveluscher thank you! 🙌
